### PR TITLE
feat(after): stabilize `unstable_after`

### PR DESCRIPTION
**Pull request by lubieowoce**
- rename `unstable_after` -> `after`
- remove `experimental.after` flag (and warn if it's present)

i separated the big find-and-replace into a separate commit so that it's easier to see what else changed apart from that.

does not touch docs, that's in #73038

### CODE REVIEW

File: docs/01-app/03-api-reference/04-functions/after.mdx
Comment (lubieowoce) [inline docs/01-app/03-api-reference/04-functions/after.mdx:1]: note to self: this is gonna need a redirect

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 1bac9fe6a967194426017509a5bb848f66c11d64

`pnpm test-dev test/development/acceptance/server-component-compiler-errors-in-pages.test.ts`
- Error Overlay for server components compiler errors in pages > importing after from 'next/server' in pages

<details>
<summary>Expand output</summary>

  ● Error Overlay for server components compiler errors in pages › importing after from 'next/server' in pages

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `Error Overlay for server components compiler errors in pages importing after from 'next/server' in pages 1`

    - Snapshot  - 2
    + Received  + 2

    @@ -1,8 +1,8 @@
      "./components/Comp.js
    - Error:   x You're importing a component that needs "after". That only works in a Server Component which is not supported in the pages/ directory. Read more: https://nextjs.org/docs/app/building-
    -   | your-application/rendering/server-components
    + Error:   x You're importing a component that needs "after". That only works in a Server Component which is not supported in the pages/ directory. Read more: https://nextjs.org/docs/app/building-your-
    +   | application/rendering/server-components
        | 
        | 
         ,-[1:1]
       1 | import { after } from 'next/server'
         :          ^^^^^

      183 |           'Import trace for requested module:'
      184 |         )
    > 185 |       ).toMatchInlineSnapshot(`
          |         ^
      186 |         "./components/Comp.js
      187 |         Error:   x You're importing a component that needs "after". That only works in a Server Component which is not supported in the pages/ directory. Read more: https://nextjs.org/docs/app/building-
      188 |           | your-application/rendering/server-components

      at Object.toMatchInlineSnapshot (development/acceptance/server-component-compiler-errors-in-pages.test.ts:185:9)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/next/src/server/config.ts [hunk 1]
```diff
@@ -453,7 +453,14 @@ function assignDefaults(
   warnOptionHasBeenDeprecated(
     result,
     'experimental.instrumentationHook',
-    '`experimental.instrumentationHook` is no longer needed to be configured in Next.js',
+    `\`experimental.instrumentationHook\` is not longer needed, because \`instrumentation.js\` is available by default. You can remove it from ${configFileName}.`,
+    silent
```
Comment (lubieowoce) [inline packages/next/src/server/config.ts:457]: rephrased this a bit while i was here, sounds a bit more natural to me

**Review by ztanner — APPROVED** [review]

**Comment by lubieowoce** [comment]
* **#73593** <a href="https://app.graphite.dev/github/pr/vercel/next.js/73593?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#73449](https://github.com/vercel/next.js/pull/73449) <a href="https://app.graphite.dev/github/pr/vercel/next.js/73449?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#73601](https://github.com/vercel/next.js/pull/73601) <a href="https://app.graphite.dev/github/pr/vercel/next.js/73601?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#73605** <a href="https://app.graphite.dev/github/pr/vercel/next.js/73605?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/vercel/next.js/73605?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by ijjk — APPROVED** [review]

**Pull request merged**